layouts/partials/components/carousel.html

```diff
@@ -0,0 +1,42 @@
+<section>
+	<div id="tinyslider-carousel">
+		{{ range sort $.Params.carousel.images "weight" }}
+		<div class="item">
+			<div>
+				<img class="img-responsive" src="{{ .image }}" alt="{{ .description }}" />
+			</div>
+			{{ if .description }}
+			<p>{{ .description }}</p>
+			{{ end }}
+		</div>
+		{{ end }}
+	</div>
+	<div class="tinyslider-controls">
+		<ul class="thumbnails" id="customize-thumbnails">
+		{{ range sort $.Params.carousel.images "weight" }}
+			<li class="item"><img class="img-responsive" src="{{ .image }}" alt="{{ .description }}" width="50px" /></li>
+		{{ end }}
+		</ul>
+		<ul class="controls" id="customize-controls">
+			{{- $angleleft := resources.Get "images/angle-left.png" }}
+			{{- $angleright := resources.Get "images/angle-right.png" }}
+			<li class="prev"><img src="{{ $angleleft.Permalink }}" alt="Move selection to left" /></li>
+			<li class="next"><img src="{{ $angleright.Permalink }}" alt="Move selection to right" /></li>
+		</ul>
+	</div>
+	<script src="https://cdnjs.cloudflare.com/ajax/libs/tiny-slider/2.9.3/min/tiny-slider.js"></script>
```

Comment:
Please do not hard code assets. You can define it in `data/assets.toml` and call it with something like `{{ printf $assets.highlightjs.js.url $assets.highlightjs.version }}`. You can refer to `partials/head.html:78` for more details. And please use jsdelivr if it is possible. This theme has many users in China and jsdelivr performs much better than cloudflare in China.

And please use `async` or `defer` for script if it is possible.

Reply:
Could not add defer or async as object is called right at site creation. When I tried the console reported tns object not defined...

Changed to usage of `assets.toml` c1e6f71